src/PowerShellEditorServices/Services/PowerShell/Execution/SynchronousTask.cs

```diff
@@ -35,7 +35,7 @@ internal abstract class SynchronousTask<TResult> : ISynchronousTask
             CancellationToken cancellationToken)
         {
             Logger = logger;
-            _taskCompletionSource = new TaskCompletionSource<TResult>();
+            _taskCompletionSource = new TaskCompletionSource<TResult>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Can we comment exactly why we're doing this? (Like what did you hit that we have to set it.)

Reply:
You can kind of think of it like `.ConfigureAwait(false)` for completion sources. I didn't necessarily need to add it here, it was just a troubleshooting step, *but* we probably should be adding it everywhere. [Related blog post](https://devblogs.microsoft.com/premier-developer/the-danger-of-taskcompletionsourcet-class/)

I can add a comment here but maybe I should just update all the other instances as well? If we go that route would you still prefer a comment over each one?

Reply:
We don't need to do it in this PR, but thanks for the blog post. I'll make an issue for further investigation.